robot/impl/local_robot.go

```diff
@@ -171,6 +171,17 @@ func (r *localRobot) Close(ctx context.Context) error {
 	return err
 }
 
+// CloseWait attempts to cleanly close down all constituent parts of the robot, including
+// reconfigureWorkers that may be running outside code and have unexpected behavior. This
+// method should only be used for testing to ensure there are no goroutine leaks.
+func (r *localRobot) CloseWait(ctx context.Context) error {
+	if err := r.Close(ctx); err != nil {
+		return err
+	}
+	r.reconfigureWorkers.Wait()
+	return nil
+}
```

Comment:
flyby: this is basically a goleak-safe version of a close, not strictly needed for this change so I can remove.

Reply:
on second thought, I will remove... the change is noisy enough as is